pkg/bootstrap/platform/gcp.go

```diff
@@ -249,33 +253,65 @@ func waitForMetadataSuppliers(suppliers []metadataSupplier, md map[string]string
 	return &wg
 }
 
+// Zones are in the form <country_code>-<region_suffix>-<zone_suffix>.
+var zoneRE = regexp.MustCompile(`^([^-]+-[^-]+)-[^-]+$`)
+
 // Converts a GCP zone into a region.
 func zoneToRegion(z string) (string, error) {
-	// Zones are in the form <region>-<zone_suffix>, so capture everything but the suffix.
-	re := regexp.MustCompile("(.*)-.*")
-	m := re.FindStringSubmatch(z)
+	m := zoneRE.FindStringSubmatch(z)
 	if len(m) != 2 {
 		return "", fmt.Errorf("unable to extract region from GCP zone: %s", z)
 	}
 	return m[1], nil
 }
 
+// Returns the zone where the pod is located in.
+func (e *gcpEnv) getPodZone() (string, error) {
+	z := e.cachedZone.Load()
+	if z != "" {
+		return z, nil
+	}
+	ctx, cfn := context.WithTimeout(context.Background(), 3*time.Second)
```

Comment:
3s is too optimistic. The library has max timout of 30s with repeated retries.  P99.9 is 5s for GKE MDS.

Reply:
Increased to 10s.